fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -239,64 +239,101 @@ def shell_escaped_password(password)
   # escaping problems in its accompanying shell script.
   class JavaTransporterExecutor < TransporterExecutor
     def build_upload_command(username, password, source = "/tmp", provider_short_name = "")
-      [
-        Helper.transporter_java_executable_path.shellescape,
-        "-Djava.ext.dirs=#{Helper.transporter_java_ext_dir.shellescape}",
-        '-XX:NewSize=2m',
-        '-Xms32m',
-        '-Xmx1024m',
-        '-Xms1024m',
-        '-Djava.awt.headless=true',
-        '-Dsun.net.http.retryPost=false',
-        java_code_option,
-        '-m upload',
-        "-u #{username.shellescape}",
-        "-p #{password.shellescape}",
-        "-f #{source.shellescape}",
-        additional_upload_parameters, # that's here, because the user might overwrite the -t option
-        '-k 100000',
-        ("-itc_provider #{provider_short_name}" unless provider_short_name.to_s.empty?),
-        '2>&1' # cause stderr to be written to stdout
-      ].compact.join(' ') # compact gets rid of the possibly nil ENV value
+      if Helper.mac? && Helper.xcode_at_least?(11)
+        [
+          'xcrun iTMSTransporter',
+          '-m upload',
+          "-u #{username.shellescape}",
+          "-p #{password.shellescape}",
+          "-f #{source.shellescape}",
+          additional_upload_parameters, # that's here, because the user might overwrite the -t option
+          '-k 100000',
+          ("-itc_provider #{provider_short_name}" unless provider_short_name.to_s.empty?),
+          '2>&1' # cause stderr to be written to stdout
```

Comment:
I'm wondering if we can get separate pipes for stdout and stderr and then treat each one individually. What do you think?

Reply:
I’m very hesitant to change that right now 😇 I’d be down for experimenting in a separate PR after 2.150.0 😊 I just want to fix things to working lol